pkg/controller/argocd/util.go

```diff
@@ -1097,10 +1097,28 @@ func containsString(arr []string, s string) bool {
 func namespaceFilterPredicate() predicate.Predicate {
 	return predicate.Funcs{
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			// If ArgoCDManagedByLabel exists, return true.
-			// Event is then handled by the reconciler.
-			if _, ok := e.MetaNew.GetLabels()[common.ArgoCDManagedByLabel]; ok {
-				return true
+			// This checks if ArgoCDManagedByLabel exists in newMeta, if exists then -
+			// 1. Check if oldMeta had the label or not? if no, return true
+			// 2. if yes, check if old and new value are the same, if yes, return true.
+			// 3. If old and new value are not the same, first deleteRBACs for old val, return true.
+			// Event is then handled by the reconciler, which would create appropriate RBACs.
+			if valNew, ok := e.MetaNew.GetLabels()[common.ArgoCDManagedByLabel]; ok {
+				if valOld, ok := e.MetaOld.GetLabels()[common.ArgoCDManagedByLabel]; !ok || valOld == "" {
```

Comment:
Can this if , if-else and else block be modified into Just one if block ?

if valOld != valNew {
      // If old and new value are not the same, first deleteRBACs for old val, return true.
     // Event is then handled by the reconciler, which would create appropriate RBACs.
}

Reply:
Thanks @iam-veeramalla, fixed it!!